### ISSUE TITLE
Version 3.5.5 has an updated archive filename 

### ISSUE THREAD

**Issue by alexconrey** [deric/puppet-zookeeper#129]
As of `3.5.5` it looks like the filename has updated for archive paths. I've got a PR incoming but wanted to tag this as a formal issue. 

# Issue
```
Error: Execution of '/bin/curl http://apache.org/dist/zookeeper/zookeeper-3.5.5/zookeeper-3.5.5.tar.gz -o /tmp/zookeeper-3.5.5.tar.gz_20190602-6793-i56h6m -fsSLg --max-redirs 5' returned 22: curl: (22) The requested URL returned error: 404 Not Found
Error: /Stage[main]/Zookeeper::Install::Archive/Archive[/opt/zookeeper-3.5.5.tar.gz]/ensure: change from 'absent' to 'present' failed: Execution of '/bin/curl http://apache.org/dist/zookeeper/zookeeper-3.5.5/zookeeper-3.5.5.tar.gz -o /tmp/zookeeper-3.5.5.tar.gz_20190602-6793-i56h6m -fsSLg --max-redirs 5' returned 22: curl: (22) The requested URL returned error: 404 Not Found
Notice: /Stage[main]/Zookeeper::Install::Archive/Exec[chown_zookeeper_directory]: Dependency Archive[/opt/zookeeper-3.5.5.tar.gz] has failures: true
```

This is legitimate, the new archive is located at: https://www.apache.org/dist/zookeeper/zookeeper-3.5.5/apache-zookeeper-3.5.5.tar.gz 

# (Proposed) Resolution
Check version specified in `$::zookeeper::archive_version` and adjust filename accordingly

```
   if versioncmp($archive_version, '3.5.5') <= 0 {
     $filename = "apache-${module_name}-${archive_version}"
   } else {
     $filename = "${module_name}-${archive_version}"
   }
```

**Comment by deric** [comment]
I've fixed tests and set mirror URI according to version (though you can still override it if you want). Let me know if you run into any problems.

**Comment by swmadmsys** [comment]
This case in `archive.pp` :
```
  # Apache updated the filename base for archive files in release 3.5.5
  if versioncmp($::zookeeper::archive_version, '3.5.5') >= 0 {
    $filename = "apache-${module_name}-${::zookeeper::archive_version}-bin"
    $archive_dl_site = $::zookeeper::archive_dl_site ? {
      undef   => 'http://apache.org/dist',
      default => $::zookeeper::archive_dl_site,
    }
  } else {
    $filename = "${module_name}-${::zookeeper::archive_version}"
    $archive_dl_site = $::zookeeper::archive_dl_site ? {
      undef   => 'http://archive.apache.org/dist/zookeeper',
      default => $::zookeeper::archive_dl_site,
    }
  }
```
seems to not be necessary.
For 3.5.5, archive is available at : [http://archive.apache.org/dist/zookeeper/zookeeper-3.5.5/](http://archive.apache.org/dist/zookeeper/zookeeper-3.5.5/)
whereas archive is not available at [http://apache.org/dist/zookeeper-3.5.5/](http://apache.org/dist/zookeeper-3.5.5/). Missing a subfolder `zookeeper`

**Comment by deric** [comment]
@swmadmsys should be fixed in d33cf0289333846775860751d12533e72f284c3e, thanks for pointing that out.
